### PR TITLE
Masq review test

### PR DESCRIPTION
**Pull request by leventdem**
Minor imporvements for tests
Import change of call order of login/register (add await also)
Add checkObject for profile/app/device

### CODE REVIEW

File: src/library/masq.js [hunk 1]
```diff
@@ -9,14 +9,30 @@ import { isUsernameAlreadyTaken } from './utils'
...
 
 const swarmOpts = process.env.NODE_ENV === 'test'
   ? { wrtc: require('wrtc') }
   : {}
 
+const requiredParametersDevice = [
+  'name'
+]
+
+const requiredParametersApp = [
```
Comment (G-Ray) [inline src/library/masq.js:24]: I think you can add appId

File: src/library/masq.js [hunk 2]
```diff
@@ -9,14 +9,30 @@ import { isUsernameAlreadyTaken } from './utils'
...
+  'name',
+  'description'
+]
+
+const requiredParametersProfile = [
+  'username',
```
Comment (G-Ray) [inline src/library/masq.js:31]: Even if image can be empty `''` we should check that it is present

File: src/library/masq.js [hunk 3]
```diff
@@ -62,7 +79,8 @@ class Masq {
       throw new MasqError(ERRORS.INVALID_PASSPHRASE)
     }
 
-    this.profileDB.on('ready', () => this._startReplicate(this.profileDB))
+    await dbReady(this.profileDB)
```
Comment (G-Ray) [inline src/library/masq.js:84]: :+1:

File: src/library/masq.js (hunk 1 above)
Comment (leventdem) [reply src/library/masq.js:24]: ok, done.

File: src/library/masq.js (hunk 2 above)
Comment (leventdem) [reply src/library/masq.js:31]: ok done

File: src/library/masq.js [hunk 4]
```diff
@@ -317,9 +340,7 @@ class Masq {
    * It will create a new hyperdb with a given id.
    * Then, the user-app send its local key to authorize.
    * This method should be called only once the user is logged in
-   * @param {string} channel The channel to connect
-   * @param {string} rawKey The encryption key, base64 encoded
-   * @param {string} appId The app id (url for instance)
+   * @param {boolean} isGranted True if the app is authorized
```
Comment (G-Ray) [inline src/library/masq.js:343]: :+1:

**Review by G-Ray — APPROVED** [review]

**Review by sujeebant — APPROVED** [review]

**Pull request merged**